src/irr/asset/CImageLoaderTGA.cpp

```diff
@@ -265,74 +308,38 @@ asset::SAssetBundle CImageLoaderTGA::loadAsset(io::IReadFile* _file, const asset
-					convertColorFlip<EF_R8_SRGB, EF_R8_SRGB>(region.bufferRowLength, region.getExtent(), texelBuffer, texelBuffer, false);
-				}
-				else {
-					// Do an OpenGL flip
-					convertColorFlip<EF_R8_SRGB, EF_R8_SRGB>(region.bufferRowLength, region.getExtent(), texelBuffer, texelBuffer, true);
-				}
-				
-				imgInfo.format = EF_R8G8B8_SRGB; // converting R8 to R8G8B8 is placed bellow
-
-				const void* planarData[] = { texelBuffer->getPointer() , nullptr, nullptr, nullptr };
-				const size_t wholeSize = region.imageExtent.height * region.bufferRowLength;
-				const auto wholeSizeInBytesAfterConvertion = wholeSize * getTexelOrBlockBytesize(EF_R8G8B8_SRGB);
-				uint8_t* outRGBData = _IRR_NEW_ARRAY(uint8_t, wholeSizeInBytesAfterConvertion);
-
-				convertColor<EF_R8_SRGB, EF_R8G8B8_SRGB>(planarData, outRGBData, wholeSize, *reinterpret_cast<core::vector3d<uint32_t>*>(&region.imageExtent));
-
-				texelBuffer = std::move(core::make_smart_refctd_ptr<ICPUBuffer>(wholeSizeInBytesAfterConvertion));
-
-				memcpy(texelBuffer->getPointer(), outRGBData, wholeSizeInBytesAfterConvertion);
-				_IRR_DELETE_ARRAY(outRGBData, wholeSizeInBytesAfterConvertion); // it involves size in R8G8B8
+				newConvertedImage = createAndconvertImageData<asset::EF_R8_SRGB, asset::EF_R8G8B8A8_SRGB>(imgInfo, texelBuffer, regions, flip);
 			}
 			break;
 		case 16:
 			{
-				imgInfo.format = asset::EF_A1R5G5B5_UNORM_PACK16;
-
-				if (flip)
-					convertColorFlip<EF_A1R5G5B5_UNORM_PACK16, EF_A1R5G5B5_UNORM_PACK16>(region.bufferRowLength, region.getExtent(), texelBuffer, texelBuffer, false);
-				else
-					convertColorFlip<EF_A1R5G5B5_UNORM_PACK16, EF_A1R5G5B5_UNORM_PACK16>(region.bufferRowLength, region.getExtent(), texelBuffer, texelBuffer, true);
+				newConvertedImage = createAndconvertImageData<asset::EF_A1R5G5B5_UNORM_PACK16, asset::EF_A1R5G5B5_UNORM_PACK16>(imgInfo, texelBuffer, regions, flip);
 			}
 			break;
 		case 24:
 			{
-				imgInfo.format = asset::EF_R8G8B8_SRGB;
-				
-				if (flip)
-					convertColorFlip<EF_B8G8R8_SRGB, EF_R8G8B8_SRGB>(region.bufferRowLength, region.getExtent(), texelBuffer, texelBuffer, false);
-				else
-					convertColorFlip<EF_B8G8R8_SRGB, EF_R8G8B8_SRGB>(region.bufferRowLength, region.getExtent(), texelBuffer, texelBuffer, true);
+				newConvertedImage = createAndconvertImageData<asset::EF_B8G8R8_SRGB, asset::EF_R8G8B8_SRGB>(imgInfo, texelBuffer, regions, flip);
 			}
 			break;
 		case 32:
 			{
-				imgInfo.format = asset::EF_R8G8B8A8_SRGB;
-
-				if (flip)
-					convertColorFlip<EF_B8G8R8A8_SRGB, EF_R8G8B8A8_SRGB>(region.bufferRowLength, region.getExtent(), texelBuffer, texelBuffer, false);
-				else
-					convertColorFlip<EF_B8G8R8A8_SRGB, EF_R8G8B8A8_SRGB>(region.bufferRowLength, region.getExtent(), texelBuffer, texelBuffer, true);
+			newConvertedImage = createAndconvertImageData<asset::EF_B8G8R8A8_SRGB, asset::EF_R8G8B8A8_SRGB>(imgInfo, texelBuffer, regions, flip);
```

Comment:
there's no conversion needed, maybe a flip at most

Reply:
nope, I can't remember but TGA is sensitive to it as far as I remember
it was previously commented it is required

Reply:
we support that texture format 100%

Reply:
I know https://github.com/buildaworldnet/IrrlichtBAW/pull/468#discussion_r410680855

Reply:
ok lets move this to https://github.com/buildaworldnet/IrrlichtBAW/pull/468#discussion_r410680855

Reply:
it's done